### PR TITLE
bugfix: S3C-2987 helper to get stream data as a JSON payload

### PR DESCRIPTION
**Pull request by jonathan-gramain**
Add a new helper function to get data from a stream as a JSON payload,
optionally validated against a joi schema.

Note: uses async/await, so updated the scality/Guidelines dependency
to please the linter

### CODE REVIEW

File: lib/stream/utils.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    return new Promise((resolve, reject) => {
+        const contentsChunks = [];
+        s.on('data', chunk => {
+            contentsChunks.push(chunk);
+        });
+        s.on('end', () => {
+            const contents = contentsChunks.join('');
+            try {
+                const parsedContents = JSON.parse(contents);
+                let validatedContents;
```
Comment (miniscruff) [inline lib/stream/utils.js]: Could just resolve(value) and resolve(parsedContents) instead of creating this variable. Doesn't matter though, either way.

**Review by miniscruff — APPROVED** [review]

**Review by rahulreddy — COMMENTED** [review]
One thing I suggest is instead of utils holding a couple of stream methods, export each method directly with its own file.
Instead of
-` lib/stream/utils.js`
```js
module.exports = {
    readJSONStreamObject,
};
```

use 
`+lib/stream/readJSONStreamObject.js`
```js
module.exports = readJSONStreamObject;
```

It's easy to discover and use the methods this way.

**Review by naren-rajendran — APPROVED** [review]

**Comment by jonathan-gramain** [comment]
@bert-e reset

**Comment by jonathan-gramain** [comment]
@bert-e reset

**Comment by miniscruff** [comment]
> One thing I suggest is instead of utils holding a couple of stream methods, export each method directly with its own file.
> Instead of
> -` lib/stream/utils.js`
> 
> ```js
> module.exports = {
>     readJSONStreamObject,
> };
> ```
> 
> use
> `+lib/stream/readJSONStreamObject.js`
> 
> ```js
> module.exports = readJSONStreamObject;
> ```
> 
> It's easy to discover and use the methods this way.

An alternative is a flat structure used by standard packages like python, go, or node. Most imports require no more than one path.
So something like:
`const { streamToObj } = require('stream')`

Even if the files are split up using index.js to manipulate and simplify the imports can be convenient. Very different from super nested structures from Java and C++ ( I think ).

**Comment by jonathan-gramain** [comment]
@bert-e approve

**Pull request merged**